apps/src/p5lab/poetry/PoemSelector.jsx

```diff
@@ -2,20 +2,95 @@
 import React, {useState} from 'react';
 import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
+import StylizedBaseDialog, {
+  FooterButton
+} from '@cdo/apps/componentLibrary/StylizedBaseDialog';
 import {setPoem} from '../redux/poetry';
 import msg from '@cdo/poetry/locale';
 import {APP_WIDTH} from '../constants';
 import {POEMS} from './constants';
 
+function PoemEditor(props) {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [poem, setPoem] = useState('');
+
+  const body = (
+    <div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.title()}: `}</label>
+        <input
+          style={styles.input}
+          value={title}
+          onChange={event => setTitle(event.target.value)}
+        />
+      </div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.author()}: `}</label>
+        <input
+          style={styles.input}
+          value={author}
+          onChange={event => setAuthor(event.target.value)}
+        />
+      </div>
+      <div style={styles.modalContainer}>
+        <label style={styles.label}>{`${msg.poem()}: `}</label>
+        <textarea
+          style={styles.input}
+          value={poem}
+          onChange={event => setPoem(event.target.value)}
+        />
+      </div>
+    </div>
+  );
+
+  const footerButton = (
+    <FooterButton
+      text={msg.save()}
+      onClick={() =>
+        props.handleClose({title, author, lines: poem.split('\n')})
+      }
+      key="confirm"
```

Comment:
nit: `key` is only needed if there are multiple instances of FooterButton (e.g., the [defaultButtons in StylizedBaseDialog](https://github.com/code-dot-org/code-dot-org/blob/346afcf0831fbe0e657bd1b744411785c5f22f56/apps/src/componentLibrary/StylizedBaseDialog.jsx#L68) have keys because there are 2 buttons)